src/python/pants_release/generate_release_announcement.py

```diff
@@ -2,18 +2,17 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 import argparse
 import json
-from pathlib import Path
 
-from pants_release.common import sorted_contributors
+from pants_release.common import VERSION_PATH, sorted_contributors
 from pants_release.git import git
 
 from pants.util.strutil import softwrap
 
 
 def announcement_text() -> str:
-    version = Path("src/python/pants/VERSION").read_text().strip()
+    version = VERSION_PATH.read_text().strip()
     cur_version_sha, prev_version_sha = git(
-        "log", "-2", "--pretty=format:%h", "src/python/pants/VERSION"
+        "log", "-2", "--pretty=format:%h", str(VERSION_PATH)
     ).splitlines(keepends=False)
```

Comment:
do we want to extract this into a method in git.py to keep all the git in one place?

Reply:
For one-off/specialised calls like this, I don't have a particular point of view. For me, I'd lean away from wrapping this sort of thing up into a "`find_two_most_recent_commits_changing_version`" function unless there were more locations that needed it! 🤷

Reply:
That's fair. There's always a tension between keeping everything at the same abstraction level and having a pile of hyper-specific functions. I find that naming the function by why I would use it helps me determine if it's worth wrapping. (Put another way: the view of this interface by the consumer rather than the provider.) something like `get_commit_filter_for_this_release` would fit more naturally in a list of steps that this function performs, like:

1. get git commit filter for this release
2. get and sort all contributors from that range
3. identify new contributors
4. ...

You're right that this would make our git.py contain a lot of one-off functions.